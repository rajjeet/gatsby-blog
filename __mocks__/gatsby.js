/* eslint no-undef: 0 */
/* eslint @typescript-eslint/no-unused-vars: 0 */
import React from 'react';
import { mockGroups } from '../src/fixtures/tags';
import { siteMetadata } from '../gatsby-config';
import { mockGatsbyImageSharpFluid } from '../src/fixtures/gatsby-images';

const gatsby = jest.requireActual('gatsby');
module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) => React.createElement('a', {
      ...rest,
      href: to,
    }),
  ),
  StaticQuery: jest.fn().mockImplementation(() => ({
    file: mockGatsbyImageSharpFluid.file,
    categoryGrouping: mockGroups.categoryGrouping,
    site: {
      siteMetadata,
    },
  })),
  useStaticQuery: jest.fn().mockImplementation(() => ({
    file: mockGatsbyImageSharpFluid.file,
    categoryGrouping: mockGroups.categoryGrouping,
    site: {
      siteMetadata,
    },
  })),
};
