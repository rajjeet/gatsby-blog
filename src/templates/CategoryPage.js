import React from 'react';
import { graphql } from 'gatsby';
import Layout from '../components/Layout';
import PostListing from '../components/PostListing';

export default ({ data, pageContext }) => (
  <Layout>
    <PostListing
      posts={data.allMarkdownRemark.edges}
      heading={pageContext.category}
      numOfPages={pageContext.numOfPages}
      currentPage={pageContext.currentPage}
      paginationSlug={pageContext.paginationSlug}
    />
  </Layout>
);

export const query = graphql`
    query($category: String!, $skip: Int!, $limit: Int!) {
      allMarkdownRemark (
        limit: $limit
        skip: $skip
        sort: {fields: frontmatter___date, order: DESC}
        filter: {frontmatter: {category: {eq: $category }, draft: {ne: true} }, fields: { contentType: { eq: "post" } } }
      ) {        
        edges{
          node {
            ...PostListingMarkdownFragment
          }
        }    
      }              
    }
`;
