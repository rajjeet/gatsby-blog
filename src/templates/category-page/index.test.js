import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import CategoryPage from './index';
import {
  createMockGatsbyImageSharpFluid,
  createMockGroups,
  createMockPosts,
} from '../../utils/testing';
import siteMetadata from '../../../gatsby-config';

const makeProps = () => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    limit: 5,
    skip: 2,
    numOfPages: 2,
    numOfPosts: 10,
    currentPage: 3,
  },
});

beforeEach(() => {
  StaticQuery.mockImplementation(({ render }) => render(
    {
      file: createMockGatsbyImageSharpFluid.file,
      site: siteMetadata,
      categoryGrouping: createMockGroups.categoryGrouping,
    },
  ));
});

describe('CategoryPage', () => {
  it('should render', () => {
    const tree = renderer.create(<CategoryPage {...makeProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
