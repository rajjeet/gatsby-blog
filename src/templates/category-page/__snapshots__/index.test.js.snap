// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage should render 1`] = `
<div
  className="sc-htpNat faJHDz"
>
  <div
    className="sc-bwzfXH bmdiOC"
  >
    <a
      href="/"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
            "width": "3em",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        
        <picture>
          <source
            sizes=""
            srcSet=""
          />
          <img
            alt=""
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
    <a
      href="/"
    >
      <h2>
        Ortmesh
      </h2>
      <p>
        Write code that matters
      </p>
    </a>
    <button
      className="sc-bdVaJa jCUGPA"
      onClick={[Function]}
    >
      About
    </button>
  </div>
  <div
    className="sc-jTzLTM juNwyP"
  >
    <div
      className="sc-gqjmRU fCAPEA"
    >
      <h1>
        Posts
      </h1>
      <div>
        <p
          style={
            Object {
              "color": "#888",
              "marginBottom": ".3em",
            }
          }
        >
          3
           
          of
           
          2
           
          Pages
        </p>
        <div
          className="sc-iwsKbI iuVdIU"
        >
          <button
            className="sc-bdVaJa jCUGPA"
            disabled={false}
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
              Newer Posts
          </button>
          <button
            className="sc-bdVaJa jCUGPA"
            disabled={true}
            onClick={[Function]}
          >
            Older Posts  
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "marginTop": ".4em",
            "padding": ".5em 1em",
          }
        }
      >
        <div
          className="post"
        >
          <div
            className="image-container"
          >
            <div
              className="sc-gZMcBi djMVKd gatsby-image-wrapper"
              style={
                Object {
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingBottom": "100%",
                    "width": "100%",
                  }
                }
              />
              
              <picture>
                <source
                  sizes=""
                  srcSet=""
                />
                <img
                  alt=""
                  onLoad={[Function]}
                  src=""
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 0.5s",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="post-summary"
          >
            <a
              href="/adding-project-section-to-website/"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
            >
              <h3
                style={
                  Object {
                    "marginBottom": "0em",
                  }
                }
              >
                Adding a Project Section to My Website
              </h3>
              <div
                style={
                  Object {
                    "color": "dimgray",
                    "fontSize": ".8em",
                    "marginTop": "0em",
                  }
                }
              >
                11 May, 2019
                 - 
                2 min read
              </div>
              <div
                style={
                  Object {
                    "fontSize": ".8em",
                  }
                }
              >
                Over the past year, I truly learned the importance of building side projects as a software developer. Besides building projects, showing them to the world is just as important. That's why I created the project section.
              </div>
            </a>
            <div
              className="sc-bxivhb iLVAZO"
            >
              <span
                onClick={[Function]}
              >
                Side Project
              </span>
            </div>
            <div
              className="sc-bxivhb iLVAZO"
            >
              <span
                onClick={[Function]}
              >
                Gatsby JS
              </span>
            </div>
          </div>
        </div>
        <div
          className="post"
        >
          <div
            className="image-container"
          >
            <div
              className="sc-gZMcBi djMVKd gatsby-image-wrapper"
              style={
                Object {
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "paddingBottom": "100%",
                    "width": "100%",
                  }
                }
              />
              
              <picture>
                <source
                  sizes=""
                  srcSet=""
                />
                <img
                  alt=""
                  onLoad={[Function]}
                  src=""
                  style={
                    Object {
                      "height": "100%",
                      "left": 0,
                      "objectFit": "cover",
                      "objectPosition": "center",
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transition": "opacity 0.5s",
                      "width": "100%",
                    }
                  }
                />
              </picture>
              <noscript
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<picture><img src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="post-summary"
          >
            <a
              href="/state-management-react-hooks/"
              style={
                Object {
                  "color": "black",
                  "textDecoration": "none",
                }
              }
            >
              <h3
                style={
                  Object {
                    "marginBottom": "0em",
                  }
                }
              >
                State Management using React Hooks
              </h3>
              <div
                style={
                  Object {
                    "color": "dimgray",
                    "fontSize": ".8em",
                    "marginTop": "0em",
                  }
                }
              >
                02 May, 2019
                 - 
                4 min read
              </div>
              <div
                style={
                  Object {
                    "fontSize": ".8em",
                  }
                }
              >
                I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.
              </div>
            </a>
            <div
              className="sc-bxivhb iLVAZO"
            >
              <span
                onClick={[Function]}
              >
                Guide
              </span>
            </div>
            <div
              className="sc-bxivhb iLVAZO"
            >
              <span
                onClick={[Function]}
              >
                React
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-iwsKbI iuVdIU"
      >
        <button
          className="sc-bdVaJa jCUGPA"
          disabled={false}
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-left fa-w-10 "
            data-icon="chevron-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
            Newer Posts
        </button>
        <button
          className="sc-bdVaJa jCUGPA"
          disabled={true}
          onClick={[Function]}
        >
          Older Posts  
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-chevron-right fa-w-10 "
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="sc-VigVT jLXPPq"
    >
      <div
        className="sc-EHOje khogYR"
      >
        <h5>
          Categories
        </h5>
        <div
          className="sc-bxivhb fXNsAS"
        >
          <span
            onClick={[Function]}
          >
            Experience
            <span>
              1
            </span>
          </span>
          <span
            onClick={[Function]}
          >
            Guide
            <span>
              2
            </span>
          </span>
        </div>
      </div>
      <div
        className="sc-ifAKCX faWeYB"
      >
        <h5>
          Tags
        </h5>
        <div
          className="sc-bxivhb fXNsAS"
        >
          <span
            onClick={[Function]}
          >
            Experience
            <span>
              1
            </span>
          </span>
          <span
            onClick={[Function]}
          >
            Guide
            <span>
              2
            </span>
          </span>
        </div>
      </div>
      <br />
      <div
        className="sc-gzVnrw eNcaSj"
      >
        <div
          className="sc-bZQynM iSfAqa gatsby-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "paddingBottom": "100%",
                "width": "100%",
              }
            }
          />
          
          <picture>
            <source
              sizes=""
              srcSet=""
            />
            <img
              alt="Author headshot"
              onLoad={[Function]}
              src=""
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 0.5s",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><img src=\\"\\" alt=\\"Author headshot\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
        <h3>
          Rajjeet Phull
        </h3>
        <p>
          Software Developer. Specializing in .NET(C#), React, SQL Server, and AWS.
        </p>
      </div>
      <div
        className="sc-dnqmqq dgVdoL"
      >
        <h4>
          Links
        </h4>
        <div>
          <span
            className="sc-htoDjs LBGjh"
          >
            <a
              href="mailto:rajjeet.phull@gmail.com?subject=Via Ortmesh:"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-envelope fa-w-16 fa-2x "
                data-icon="envelope"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </span>
          <span
            className="sc-htoDjs LBGjh"
          >
            <a
              href="https://github.com/rajjeet"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-github fa-w-16 fa-2x "
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </span>
          <span
            className="sc-htoDjs LBGjh"
          >
            <a
              href="https://www.linkedin.com/in/rajjeetphull/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-linkedin fa-w-14 fa-2x "
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </span>
          <span
            className="sc-htoDjs LBGjh"
          >
            <a
              href="https://www.facebook.com/ortmesh/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-facebook fa-w-16 fa-2x "
                data-icon="facebook"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </span>
          <span
            className="sc-htoDjs LBGjh"
          >
            <a
              href="https://twitter.com/ortmesh"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter fa-w-16 fa-2x "
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
